BuildScript.Nuke/Build.ImportConfiguration.cs

```diff
@@ -0,0 +1,115 @@
+using Nuke.Common.ProjectModel;
+using Project = Microsoft.Build.Evaluation.Project;
+
+
+internal partial class Build : NukeBuild
+{
+  private const string c_projectsFileName = "Projects.props";
+  private const string c_propertiesFileName = "Properties.props";
+  private const string c_buildConfigFileName = "BuildConfiguration.props";
+  private const string c_normalTestConfigurationProperty = "NormalTestConfiguration";
+  private const string c_releaseProjectFilesItem = "ReleaseProjectFiles";
+  private const string c_unitTestProjectFilesItem = "UnitTestProjectFiles";
+  private const string c_assemblyInfoFileProperty = "AssemblyInfoFile";
+  private const string c_companyNameProperty = "CompanyName";
+  private const string c_companyUrlProperty = "CompanyUrl";
+  private const string c_copyrightProperty = "Copyright";
+  private const string c_productNameProperty = "ProductName";
+  private const string c_supportedTargetRuntimesProperty = "SupportedTargetRuntimes";
+  private const string c_supportedExecutionRuntimesProperty = "SupportedExecutionRuntimes";
+  private const string c_versionFileName = "Version.props";
+  private const string c_versionProperty = "Version";
+  private const string c_testConfigurationMetaData = "TestConfiguration";
+  private const string c_solutionDirectoryPropName = "SolutionDirectory";
+  private string NormalTestConfiguration { get; set; } = "";
+  private string AdditionalBuildMetadata { get; } = "";
+  private IEnumerable<string> SupportedTargetRuntimes { get; set; } = new List<string>();
+  private IEnumerable<string> SupportedExecutionRuntimes { get; set; } = new List<string>();
+
+
+  public Target ImportDefinitions => _ => _
+      .Executes(() =>
+      {
+        ImportProjectDefinition();
+        ImportPropertiesDefinition();
+        ImportBuildConfigurationDefinition();
+        ImportVersion();
+      });
+
+  private void ImportProjectDefinition ()
+  {
+    var xmlProperties = PrepareGeneralProperties(c_projectsFileName);
+    var normalTestConfiguration = xmlProperties.Properties.Single(prop => prop.Name == c_normalTestConfigurationProperty);
+    var releaseProjectFiles = xmlProperties.Items.Where(prop => prop.ItemType == c_releaseProjectFilesItem);
+    var unitTestProjectFiles = xmlProperties.Items.Where(prop => prop.ItemType == c_unitTestProjectFilesItem);
+
+    NormalTestConfiguration = normalTestConfiguration.EvaluatedValue;
+    ReleaseProjectFiles = releaseProjectFiles.Select(
+        x => SetupReleaseProjectMetadata(x.EvaluatedInclude)).ToList();
+    UnitTestProjectFiles = unitTestProjectFiles.ToDictionary(
+        x => (AbsolutePath) x.EvaluatedInclude,
+        x => x.GetMetadataValue(c_testConfigurationMetaData));
+  }
+
+  private ReleaseProjectMetadata SetupReleaseProjectMetadata (string path)
+  {
+    var xmlProperties = ProjectModelTasks.ParseProject(path);
+    var isMultiTargetFramework = xmlProperties.Properties.Any(x => x.Name == "TargetFrameworks");
```

Comment:
What happens if only one target framework is set using the `TargetFrameworks`? Is `DispatchToInnerBuilds` needed then?